client/src/components/products/ProductsList.js

```diff
@@ -86,13 +82,23 @@ class ProductsList extends Component {
       )
     ];
 
+    const distinctCity = [
+      ...new Set(
+        this.state.products.map(product => {
+          return product.location.city;
+        })
+      )
+    ];
+
     const maxPrice = Math.max(
       ...this.state.products.map(product => {
         return product.price;
       })
     );
 
     let filteredProduct = this.state.products.filter(product => {
+      let isSoldMatch = product.isSold === false;
```

Comment:
you might need () ...?
let isSoldMatch = (product.isSold === false)

Reply:
changed it :)